libiocage/lib/Release.py

```diff
@@ -517,6 +496,51 @@ def update(self):
 
         return changed
 
+    def snapshot(
+        self,
+        identifier: str,
+        force: bool=False
+    ) -> libzfs.ZFSSnapshot:
+        """
+        Returns a ZFS snapshot of the release
+
+        Args:
+
+            identifier:
+                This string specifies the snapshots name
+
+            force: (default=False)
+                Enabling this option forces re-creation of a snapshot in case
+                it already exists for the given idenfifier
+
+        Returns:
+
+            libzfs.ZFSSnapshot: The ZFS snapshot object found or created
+        """
+
+        snapshot_name = f"{self.resource.dataset.name}@{identifier}"
+
+        try:
+            existing_shapshot = self.zfs.get_snapshot(snapshot_name)
+            if force is False:
+                self.logger.verbose(
+                    f"Re-using release snapshot {self.name}@{identifier}"
+                )
+                return existing_shapshot
+        except libzfs.ZFSException:
```

Comment:
We should be catching the actual zfs error code to make sure it does indeed exists.

```python
except libzfs.ZFSException as e:
    if e.code != libzfs.Error.THEONEWEWANT:
       raise
    existing_snapshot = None
```

Reply:
Indeed, but I would like to populate this pattern in the whole library at once.